### ISSUE TITLE
Deployments managed by InferenceService are not updating status at all!

### ISSUE THREAD

**Issue by abdulvirta** [kserve/kserve#2276]
/kind bug

**What steps did you take and what happened:**
kubectl rollout status deployment/my-service-transformer-default -n my-namespace

It keeps hanging with this response:
Waiting for deployment spec update to be observed...

This deployment was created by Inference Service. I'm using RawDeployment

**What did you expect to happen:**
It should tell me the status of the deployment. That it has successfully finished or something like that. This command works for any other deployment on my cluster not managed by InferenceService Controller. But for deployments created by InferenceService it just hang.

**Anything else you would like to add:**
I just feel there is a bigger and more general problem here. From two days investigating this issue, I just feel InferenceService is not really updating the status of deployments and something seems to be fundamentally wrong with these deployments. Even command like:
"kubectl rollout restart deployment/my-service-transformer-default -n my-namespace" does absolutely nothing. Not even restarting the pods!!

**Environment:**

- Istio Version:
- Knative Version: Not used (I'm using RawDeployment)
- KFServing Version: KServe 0.8
- Kubeflow version: N/A (use KServe standalone)
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version: N/A
- Kubernetes version: (use `kubectl version`): 1.23
- OS (e.g. from `/etc/os-release`): 12.4

**Comment by yuzisun** [comment]
@abdulvirta the deployment should be managed by high level resource inference service, you should never manipulate the deployment directly. This is fundamentally how KServe works, you update inference service and then kserve reconciles the deployment, if you want to check the deployment status you can check the  inference service status instead.

**Comment by abdulvirta** [comment]
Ok. Let me check the status of isvc and see if that gets updated correctly during a new rollout. Although from what I observed so far, conditions on isvc will still show as Ready. I want something to be able to use synonymous to this:
kubectl rollout status deploment/foo.

which waits until the new rollout is done. But thanks for the hint, let me try to see if I can have a similar functionality based on some condition/status of the isvc

**Comment by abdulvirta** [comment]
yeah this won't work. The conditions on inference services are not updated during a rollout. TransformerReady and PredictorReady will always have True even when a rollout is in progress. What I want is some kind of condition to wait on until a new deployment/rollout finished successfully. Like what you ideally would do with this command:
kubectl rollout status deployment my-deployment

The scenario is:
- kubectl apply ....
- Wait until the new changes have fully manifested (meaning new pods containing the new changes are up and running, and old pods are terminated)
- Do validation for the deployment.

I don't see any possible way to do this. Because when I make a new deployment, InferenceService is not keeping track on that. And the underlying deployments resources doesn't seem to get updated so doing  kubectl rollout status deployment ... command will just hang forever

**Comment by yuzisun** [comment]
@abdulvirta the ready condition is not always true, it transits to false when you make a new deployment  and marks true until the new deployment is ready.

**Comment by abdulvirta** [comment]
hmmm this is not what I'm observing for sure for whatever reason
